src/main/java/com/exasol/adapter/jdbc/BaseTableMetadataReader.java

```diff
@@ -72,7 +70,7 @@ private List<TableMetadata> extractTableMetadata(final ResultSet remoteTables, f
         do {
             final Optional<TableMetadata> tableMetadata = getTableMetadata(remoteTables, filteredTables);
             tableMetadata.ifPresent(mappedTables::add);
-            validateMappedTablesListSize(mappedTables);
+            TableCountLimit.from(this.properties).validateNumberOfTables(mappedTables.size());
```

Comment:
Move from() part outside the loop, assuming properties do not change.

Reply:
👍